### PR TITLE
use precomputed norms for raft brute_force knn calls

### PR DESCRIPTION
**Pull request by benfred**

### CODE REVIEW

**Comment by mdouze** [comment]
could you give some context?

**Comment by mdouze** [comment]
Would it be possible to support this for incremental adds?

**Comment by benfred** [comment]
> Would it be possible to support this for incremental adds?

@mdouze  This is already supported here in the `RaftFlatIndex` class here with this change.  This class subclasses the `FlatIndex` class - that supports incremental adds, and keeps the precomputed norms for the index up to date when new data is added to the index via the `FlatIndex::add` code.   We're using these precomputed norms here - and this functionality is being tested here https://github.com/facebookresearch/faiss/blob/9bb6b4be0d95042d10beba65706cacc234592128/faiss/gpu/test/test_raft.py#L55-L63

**Comment by mdouze** [comment]
Thanks for the clarification! Importing to internal version...

File: faiss/gpu/GpuDistance.cu [hunk 1]
```diff
@@ -236,89 +236,84 @@ void bfKnn(GpuResourcesProvider* prov, const GpuDistanceParams& args) {
         raft::device_resources& handle = res->getRaftHandleCurrentDevice();
         auto stream = res->getDefaultStreamCurrentDevice();
 
-        idx_t dims = args.dims;
-        idx_t num_vectors = args.numVectors;
-        idx_t num_queries = args.numQueries;
+        int64_t dims = args.dims;
```
Comment (algoriddle) [inline faiss/gpu/GpuDistance.cu:239]: @benfred Can you clarify why you're changing these to int64_t?

**Pull request closed without merging**